### PR TITLE
Major 2.0

### PR DESCRIPTION
**Pull request by XFox111**
Implements following issues: #46, #55

### Core
- Added cloud storage for sets
### UI/UX
- WeblySleek UI font replaced with Segoe fonts
- Reworked icons (#55)
- Removed changelog auto-opening on install
- 'Contributors' option replaced with 'Changelog' one
- Added icons to menu items
- Updated UI with Fluent UI compliance
- Added tooltip on how to use selective activation (#57)
### Localization
- Added Italian language (#52) by @blackcat-917
- Added Brazilian Portuguese language (#53) by @amandafilizola
### Other changes
- Updated Firefox CI/CD config
- Updated README: rearranged badges and added demo GIF
- Updated documentation

## PR Checklist
- [x] Change extension version in the manifest

### CODE REVIEW

File: js/background.js [hunk 1]
```diff
@@ -290,18 +258,18 @@ function RestoreCollection(collectionIndex, removeCollection)
...
 
-	UpdateTheme();
+	chrome.browserAction.setBadgeText({ text: collections.length < 1 ? "" : collections.length.toString() });
 }
 
 function RemoveTab(collectionIndex, tabIndex)
 {
 	var set = collections[collectionIndex];
-	if (--set.tabsCount < 1)
+	if (set.tabsCount < 2)
```
Comment (AmineI) [inline js/background.js:267]: Why don't you decrement the tabsCount anymore ? That means if you have 5 tabs in a collection, and delete one, the `tabsCount` will still be 5 even though one is gone :/ ? Or am I missing an update of that count somewhere ?

**Comment by AmineI** [comment]
I have just tested the 1.9 -> migration process again to ensure no data loss, (even with abonormally high numbers of tabs they just get saved to the bookmarks instead)

I also re-tested the 2.0 release to ensure sync worked properly, on my 2 Edge Dev machines : both machines' pre-2.0 tabs were merged in the synced data as expected, and (during my testing) sync appeared in real time on the other machine for any operation (setting tabs aside, deleting one tab, or deleting a tabs aside group) done on one of them.

The only strange thing I noticed was that deleting a tab group in one machine didn't "refresh" an open pane in the other machine (even though the data and the badge updated to the correct numbers, I had to close and reopen the pane to see the group disappear. This was not needed for other operations (like deleting a tab from a group). 

I think it works really well, and the new 2.0 theme is quite pleasing too ! Great work :) !!

**Comment by XFox111** [comment]
@AmineI cool
> The only strange thing I noticed was that deleting a collection in one machine didn't "refresh"...

That's ok, just minor bug

Btw, is it supposed not to work? I've used `major-2.0` source code for both instances and it just doesn't sync data
image

**Comment by AmineI** [comment]
I hope you've been well since last time :) !
 
> > The only strange thing I noticed was that deleting a collection in one machine didn't "refresh"...
> 
> That's ok, just minor bug
Found the cause, fixed it - sending a PR - the code to reload the panel was in an if case :) 

> Btw, is it supposed not to work? I've used `major-2.0` source code for both instances and it just doesn't sync data

If you used an unpackaged version, yes, those can't use sync servers - you have to package them in a crx and add the crx id to the ExtensionInstallAllowlist in the windows registry, like I mentioned [here](https://github.com/XFox111/TabsAsideExtension/pull/43#pullrequestreview-487018502)

I can give you my regedit script and crx if you want, otherwise i'll just do a quick image tutorial to show you how to do that yourself

**Comment by AmineI** [comment]
### Mini tutorial to test sync on an extension not in the store : 

1. Pack the source in a crx file : image
(if this does not appear, enable developer mode in the extensions page)

2. The generated crx file will be in the folder where the source folder is located. Drag and drop it in the extensions page to install it.
3. Once installed, note its id on the extension page
image
 
4. Create a file with this inside (replacing the id with your own)
 ```
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Edge\ExtensionInstallAllowlist]
"1"="jpnjdghnjpicdjgiebohcgmkeneppmha"
 ```
 save it as `any_name.reg` 
5. Then double click that .reg file and accept the prompts to add its values to the windows registry. 
6. Restart Edge. 
 
The extension should be working, and will sync to any other Edge instance that have installed a crx with the same id (ie the same crx file, or a crx file signed with the same private key).

**Comment by XFox111** [comment]
Yay, we did it guys! 🎉🎉🎉 It was a helluva long PR and there are even some changes which were supposed to be here, but still. Thx everyone for your support and contribution! ❤ Especially, @AmineI

**Comment by AmineI** [comment]
You're welcome, many thanks to you as well 😊 !!

**Pull request merged**